### ISSUE TITLE
[FEAUTURE] Swipe up on app icon to open another app...

### ISSUE THREAD

**Issue by ikbb** [Deletescape-Media/Lawnchair#2787]
In previous builds, there was a feature where you can swipe up on a camera icon, for example, and set the swipe up  to open photos app.  It was really neat and handy.
Please bring it back.

**Comment by tigattack** [comment]
I'd love to see this, definitely the main thing I miss after switching from Nova!

**Comment by themooleman** [comment]
> I'd love to see this, definitely the main thing I miss after switching from Nova!

Same here!

**Comment by thayerw** [comment]
Just adding my support for this feature, as it doubles the amount of applications that can be launched from a single group of icons. If there were a swipe up/down function, similar to Nova, that would be even better. 

For anyone else who needs the feature, it's still available as Lawnchair v2 on the Play Store. While it hasn't been updated since 2019, it remains very stable.

**Comment by shvchk** [comment]
🪄✨

**Comment by owenlejeune** [comment]
Came here to request this feature and saw you beat me to it. I've been missing this since I switched from the previous versions of Lawnchair the Alpha track. Please bring it back!

**Comment by shvchk** [comment]
🪄✨

**Comment by bitokulo** [comment]
Nice feature

**Comment by Mitezuss** [comment]
when?

**Comment by themooleman** [comment]
Still holding out hope for this feature...

**Comment by kieraneglin** [comment]
Why was this reverted? This is the only thing keeping me on an older build

**Comment by Mitezuss** [comment]
@kieraneglin 

hi, what version exactly ? compatible with A13 ?

**Comment by kieraneglin** [comment]
@Mitezuss here's a screenshot - I can't tell exactly but I hope this helps. I recall it working on 13 but it's quite buggy on 14. 

![Screenshot_20231225-163552.png](https://github.com/LawnchairLauncher/lawnchair/assets/569917/8c05a966-9522-4c6e-bdf9-c429826c7a5d)

**Comment by Mitezuss** [comment]
> @Mitezuss here's a screenshot - I can't tell exactly but I hope this helps. I recall it working on 13 but it's quite buggy on 14.
> 
> ![Screenshot_20231225-163552.png](https://private-user-images.githubusercontent.com/569917/292784421-8c05a966-9522-4c6e-bdf9-c429826c7a5d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTIiLCJleHAiOjE3MDM4MDY1MzMsIm5iZiI6MTcwMzgwNjIzMywicGF0aCI6Ii81Njk5MTcvMjkyNzg0NDIxLThjMDVhOTY2LTk1MjItNGM2ZS1iZGY5LWM0Mjk4MjZjN2E1ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjMxMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyOFQyMzMwMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNzIxNzRiNTgyNjViMDkyMmIxM2JjOWYxNzEwZGI4ZTc1OWY1YzQ4MjZjN2Y3NTMzZDljNDU2Y2RjOWMzMDRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3RgSc9xXyF1AKTzX7kDNth1DzJZ-al95gkRtOmF2cGM)


The image is private... can not see
Can share a link to download or check the version on Android Settings > Apps

**Comment by Saarsk** [comment]
This is one of the most important features in my book.

I use Nova currently (but I want to move to an open source launcher). I've made my homescreen with up+down swipe gestures for each icon (and folders), along with custom icons to represent the up+down swipe apps.

This cuts down the number of icons to 1/3rd and makes it much easier to remember where stuff is. I can also keep all the icons within the reach of my thumb when I hold my phone with one hand.

For instance, in my dock I have a chat app, if I swipe up I open phone app and if I swipe down I open SMS app. I group similar or relevant apps into one icon. It's game-changing!

(Nova also supports the creation of icons from intents/actions, so by long-pressing the Brave Browser icon I get a "new tab" tile I can drag out.)

**Comment by themooleman** [comment]
Couldn't agree more. It allows you to launch apps so much more quickly and
with fewer touches. I couldn't go back to a launcher without support for
this.

**Comment by kieraneglin** [comment]
(commenting to keep the issue alive)

**Comment by flipside101** [comment]
Agreed having the swipe option on home page and dock icons at least would be amazing. up and down swipe please

**Comment by apottere** [comment]
With the uncertain future of Nova, I'd love to see this added to lawnchair.  It's the only feature I'd really miss if I switched.

**Comment by BenjenJones** [comment]
Would really love to be able to open the first app on touch on the folder and open the folder on swipe up. This is the only thing missing from Nova

**Comment by AnuupRao** [comment]
This is the only reason I'm still stuck using nova launcher. All 7 of my dock icons are folder each with categorised essential. 
Please bring this feature back from the older lawnchair 🙏🏻

**Comment by ChrisFhey** [comment]
Same here. I wanted to switch to Lawnchair today, but I seriously miss this functionality so I’ll stick to Nova for a while longer. Please consider adding this feature.

**Comment by dragibusnoir** [comment]
I would really love to see this feature implemented as well.
And using the icon of the first app in the folder as the folder icon is also very important.

**Comment by thayerw** [comment]
Thank you, @MrSluffy! Looking very forward to seeing this return from v2!

**Comment by Saarsk** [comment]
Does this also include swipe down gesture support? :)

**Updated:**
Had a peek at the commit. There'll be support for up, down, left and right gestures! That's awesome! 😊 Thanks to the developers. Looks like it's time to ditch Nova Launcher! 🤔🥳 And look for a donation link.


Perhaps in the future there'll be an automatically generated icons corresponding to the assigned gestures. Currently, I'm making my own custom icons. It doesn't allow for easy reconfiguration of gesture config and is pretty tedious.

**Comment by MrSluffy** [comment]
not yet, it will be added later swipe down and up

**Comment by Saarsk** [comment]
> not yet, it will be added later swipe down and up

I see. 😊 So there's some more work remaining, that's alright.

Thanks a lot for this! 🚀

**Comment by RaffaeleCanale** [comment]
I see an initial PR was merged, but I don't see that feature yet in the latest build (unless I'm not looking at the right place).

Any idea of when we'll be able to start using the initial implementation?

**Comment by skridofly** [comment]
the pr has been reverted as it caused some bugs with other gestures.
Because of this reverting, I think this feature request should be opened again.

**Comment by SuperDragonXD** [comment]
The feature hasn't been reverted, but hidden inside the `three-dots menu > Experimental features`. It was hidden due to a bug of not being able to move between home pages when swiping over icons.

**Comment by ghost** [comment]
Currently the experimental feature is for swiping left/right on an icon. I only have one home page but I can't get the gestures to work. Would changing it to swiping up/down interfere with the gestures for opening the notification tray and the app drawer?

**Comment by creeve4** [comment]
Why was this request closed? Is this functionality not planned?
